Model/SourcesRepository.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace Ampersand\DisableStockReservation\Model;
+
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Ampersand\DisableStockReservation\Model\SourcesFactory;
+use Ampersand\DisableStockReservation\Model\ResourceModel\Sources;
+use Ampersand\DisableStockReservation\Model\Sources as SourceModel;
+use Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionResultInterfaceFactory;
+use Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionResultInterface;
+use Ampersand\DisableStockReservation\Service\SourcesConverter;
+use Ampersand\DisableStockReservation\Model\ResourceModel\Sources\CollectionFactory;
+
+/**
+ * Class SourcesRepository
+ * @package Ampersand\DisableStockReservation\Model
+ */
+class SourcesRepository
+{
+    /**
+     * @var SourcesFactory
+     */
+    protected $sourcesFactory;
+
+    /**
+     * @var Sources
+     */
+    protected $sourcesResourceModel;
+
+    /**
+     * @var SourceSelectionResultInterfaceFactory
+     */
+    private $sourceSelectionResultFactory;
+
+    /**
+     * @var SourcesConverter
+     */
+    private $sourcesConverter;
+
+    /**
+     * @var CollectionFactory
+     */
+    private $collectionFactory;
+
+    /**
+     * SourcesRepository constructor.
+     * @param SourcesFactory $sourcesFactory
+     * @param Sources $sourcesResourceModel
+     * @param SourceSelectionResultInterfaceFactory $sourceSelectionResultFactory
+     * @param SourcesConverter $sourcesConverter
+     * @param CollectionFactory $collectionFactory
+     */
+    public function __construct(
+        SourcesFactory $sourcesFactory,
+        Sources $sourcesResourceModel,
+        SourceSelectionResultInterfaceFactory $sourceSelectionResultFactory,
+        SourcesConverter $sourcesConverter,
+        CollectionFactory $collectionFactory
+    ) {
+        $this->sourcesFactory = $sourcesFactory;
+        $this->sourcesResourceModel = $sourcesResourceModel;
+        $this->sourceSelectionResultFactory = $sourceSelectionResultFactory;
+        $this->sourcesConverter = $sourcesConverter;
+        $this->collectionFactory = $collectionFactory;
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return SourceModel
+     * @throws NoSuchEntityException
+     */
+    public function getByOrderId(string $orderId): SourceModel
+    {
+        /** @var SourceModel $sourcesModel */
+        $sourcesModel = $this->sourcesFactory->create();
+        $this->sourcesResourceModel->load(
+            $sourcesModel,
+            $orderId,
+            'order_id'
+        );
+
+        try {
+            $sourcesModel->getId();
```

Comment:
Not quite, `getId` will always return something/null (and not throw an error) due to the model extending Magento's `DataObject`. The `DataObject`'s magic `__call` will convert any getter or setter function that doesn't exist into `getData` or `setData`... So in this case `getId` could be an ID or it could be `null`, either way your exception would not be thrown.

Maybe check if `getId` is not `null`/falsey and then throw the error?